packages/core/src/core.ts

```diff
@@ -1968,6 +2000,16 @@ class ThresholdKey implements ITKey {
   private async initializeModules() {
     return Promise.all(Object.keys(this.modules).map((x) => this.modules[x].initialize()));
   }
+
+  private async computeAccountNonce(index: number) {
+    // generation should occur during tkey.init, fails if accountSalt is absent
+    this._accountSalt = this._accountSalt || (await this.getTKeyStoreItem(TSS_MODULE, "accountSalt")).value;
```

Comment:
should getTkeyStore during reconstruct and assign to the accountSalt ( only get from tkey store if the accountSalt is undefined )
- this will keep this function as sync function